Packs/PhishingURL/Scripts/DBotUpdateLogoURLPhishing/DBotUpdateLogoURLPhishing.py

```diff
@@ -0,0 +1,290 @@
+        return
+
+    if display_logos:
+        exist, _, _ = oob_model_exists_and_updated()
+        if exist:
+            model = load_demisto_model()
+        else:
+            model = load_model_from_docker()
+        display_all_logos(model)
+        return
+
+    if (action == KEY_ADD_LOGO) and (not logo_image_id or not logo_name):
+        return_error(MSG_EMPTY_NAME_OR_URL)
+    elif not logo_name:
+        return_error(MSG_EMPTY_LOGO_NAME)
+
+    if action==KEY_ADD_LOGO:
+        res = demisto.getFilePath(logo_image_id)
+        path = res['path']
+        with open(path, 'rb') as file:
+            logo_content = file.read()
+
+    exist, demisto_major_version, demisto_minor_version = oob_model_exists_and_updated()
+
+    # Case 1: model in demisto does not exist OR new major released but no logo were added -> load from docker
+    if not exist or (demisto_major_version < MAJOR_VERSION and demisto_minor_version == MINOR_DEFAULT_VERSION):
+        model = load_model_from_docker()
+        if action == KEY_ADD_LOGO:
+            success, msg = model.add_new_logo(logo_name, logo_content, associated_domains)
+        elif action == KEY_REMOVE_LOGO:
+            success, msg = model.remove_logo(logo_name)
+        else:
+            success, msg = model.update_domain_for_custom_logo(logo_name, associated_domains)
+        msg_list.append(msg)
+        if success:
+            minor, model = get_minor_version_upgrade(model)
+            msg = save_upgraded_version_model(model)
+            msg_list.append(msg)
+        else:
+            return_error(msg)
+
+    # Case where there were new new model release -> load model from demisto
+    elif (demisto_major_version == MAJOR_VERSION):
+        model = load_demisto_model()
+        if action == KEY_ADD_LOGO:
+            success, msg = model.add_new_logo(logo_name, logo_content, associated_domains)
+        elif action == KEY_REMOVE_LOGO:
+            success, msg = model.remove_logo(logo_name)
+        else:
+            success, msg = model.update_domain_for_custom_logo(logo_name, associated_domains)
+        msg_list.append(msg)
+        if success:
+            minor, model = get_minor_version_upgrade(model)
+            msg = save_upgraded_version_model(model)
+            msg_list.append(msg)
+        else:
+            return_error(msg)
+
+    # Case where new model release and logo were added -> transfer logo from model in demisto to new model in docker
+    elif (demisto_major_version < MAJOR_VERSION) and (demisto_minor_version > MINOR_DEFAULT_VERSION):
+        model_docker = load_model_from_docker()
+        model_demisto = load_demisto_model()
+        model_docker.logos_dict = model_demisto.logos_dict
+        model_docker.custom_logo_associated_domain = model_demisto.custom_logo_associated_domain
+        model_docker.top_domains = model_demisto.top_domains
+        msg_list.append(MSG_TRANSFER_LOGO)
+        if action == KEY_ADD_LOGO:
```

Comment:
again seems like same logic - line 266-280 - use function you will create

Reply:
not exactly the same but I created 1 function